### PR TITLE
Yet another round of battling building and testing containers on Windows

### PR DESCRIPTION
**Pull request by rfay**
## The Problem/Issue/Bug:

We seem to never be able to build and test our containers on WIndows *and* elsewhere. 

Trying again.

## How this PR Solves The Problem:

Script and docs (mostly) updates, and agent debugging

## Manual Testing Instructions:

Build should run reliably on each testbot, especially windows testbots.

### CODE REVIEW

**Comment by rfay** [comment]
Much pain, much learning. I think this is ready for a quick review. Mostly I learned about agent setup, and may still have some work to do on that, maybe an agent qualifier script that makes sure git works right and all the right tools are installed?

File: .buildkite/test_containers.sh [hunk 1]
```diff
@@ -7,6 +7,12 @@ set -o pipefail
 set -o nounset
 set -x
 
+echo "--- Cleanup docker"
+echo "Warning: deleting all docker containers and deleting ~/.ddev/Test*"
```
Comment (andrewfrench) [inline .buildkite/test_containers.sh]: I don't think ~/.ddev/Test* gets deleted, if we're still expecting that to happen.

Comment (rfay) [reply .buildkite/test_containers.sh]: Great point - It's a copy-from-original-script problem. This script doesn't ever create anything in ~/.ddev, I'll fix that.

**Comment by rfay** [comment]
I'm probably good with this now, my energy may have run its course again :)

**Pull request merged**